android/src/main/java/org/reactnative/camera/RNCameraView.java

```diff
@@ -508,11 +534,28 @@ public void setGoogleVisionBarcodeMode(int barcodeMode) {
     mGoogleVisionBarCodeMode = barcodeMode;
   }
 
-  public void onBarcodesDetected(WritableArray barcodesDetected) {
+  public void onBarcodesDetected(WritableArray barcodesDetected, int width, int height, byte[] imageData) {
     if (!mShouldGoogleDetectBarcodes) {
       return;
     }
-    RNCameraViewHelper.emitBarcodesDetectedEvent(this, barcodesDetected);
+
+    // See discussion in https://github.com/react-native-community/react-native-camera/issues/2786
+    final byte[] compressedImage;
```

Comment:
is this logic the same as above?

should we extract this to a method/function?

Reply:
Yes, similar logic but different barcode reader providers in Android.